### PR TITLE
checking array if empty to remove redundant flag

### PR DESCRIPTION
**Pull request by arthur2308**
bumping version **4.2.0**, regarding PR **69** 
@kav @pshanoop

### CODE REVIEW

File: charts/verdaccio/templates/deployment.yaml [hunk 1]
```diff
@@ -67,7 +67,7 @@ spec:
             - mountPath: /verdaccio/storage
               name: storage
               readOnly: false
-            {{- if .Values.useSecretHtpasswd }}
+            {{- if .Values.secrets.htpasswd }}
```
Comment (juanpicado) [inline charts/verdaccio/templates/deployment.yaml:70]: if someone is already using this, to me, looks as a breaking change.

Comment (arthur2308) [reply charts/verdaccio/templates/deployment.yaml:70]: yes, flag is deprecatedd, and version bumped to 4.3.0 
Thanks for the catch

**Comment by pshanoop** [comment]
@arthur2308 
Thanks for the PR, I always thought empty array would return `true` on helm template. I had to check [docs](https://helm.sh/docs/chart_template_guide/control_structures/#ifelse)

Can you also update the readme [here](https://github.com/verdaccio/charts#generate-htpasswd-using-helm), please?

**Comment by arthur2308** [comment]
> @arthur2308
> Thanks for the PR, I always thought empty array would return `true` on helm template. I had to check [docs](https://helm.sh/docs/chart_template_guide/control_structures/#ifelse) to make sure :)
> 
> Can you also update the readme [here](https://github.com/verdaccio/charts#generate-htpasswd-using-helm), please?

added, included the link to helm's list evaluation doc there as well.

File: README.md [hunk 2]
```diff
@@ -150,9 +150,11 @@ $ helm install npm -f values.yaml verdaccio/verdaccio
...
-config uses this.
+config uses this. The conditional statement `{{- if .Values.secrets.htpasswd }}` 
+is evaluated as false if the list is an empty collection. 
+(Source [helm flow control](https://helm.sh/docs/chart_template_guide/control_structures/#ifelse))
 
-> **Tip**: These values are in plaintext. So don't forget to put aditional
+> **Tip**: These values are in plaintext. So don't forget to put additional
 > encryption.
 
 #### Example
```
Comment (pshanoop) [inline README.md:160]: You missed the example part. You can remove line 162.

**Review by pshanoop — CHANGES_REQUESTED** [review]

**Review by pshanoop — APPROVED** [review]

**Review by juanpicado — APPROVED** [review]

**Pull request merged**